sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaTestResourceScram.java

```diff
@@ -0,0 +1,100 @@
+package com.salesforce.kafka.test;
+
+import com.salesforce.kafka.test.junit4.SharedKafkaTestResource;
+import com.salesforce.kafka.test.junit4.SharedZookeeperTestResource;
+import com.salesforce.kafka.test.listeners.BrokerListener;
+import com.salesforce.kafka.test.listeners.SaslScramListener;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import kafka.admin.ConfigCommand;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Creates and starts up a {@link SharedKafkaTestResource} that uses a custom {@link
+ * SharedZookeeperTestResource} with installed SCRAM-SHA-* credentials.
+ *
+ * <p>See {@link SharedKafkaTestResource} for usage.
+ */
+public class SharedKafkaTestResourceScram extends SharedKafkaTestResource {
+  private final SharedZookeeperTestResource sharedZookeeperTestResource =
+      new SharedZookeeperTestResource();
+
+  @Override
+  protected void setKafkaCluster(KafkaCluster kafkaCluster) {
+    // we need to replace the zkTestServer in the kafka cluster in order to change the Zookeper
+    // configuration after it started and before the kafka broker starts.
+    if (kafkaCluster instanceof KafkaTestCluster) {
+      try {
+        Field f = KafkaTestCluster.class.getDeclaredField("zkTestServer");
+
+        // the field is private, so we need to make it accessible
+        f.setAccessible(true);
+
+        // the field is final, so we need to remove the modifier
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(f, f.getModifiers() & ~Modifier.FINAL);
+
+        // use the already started (and configured) zookeeper server.
+        f.set(kafkaCluster, sharedZookeeperTestResource.getZookeeperTestServer());
+      } catch (NoSuchFieldException | IllegalAccessException e) {
+        // empty
```

Comment:
I think we should rethrow the exceptions as RuntimeException here with an informational message so that we know when the implementation of the modified `KafkaTestCluster` changes.

Reply:
I guess when they change the implementation the Kafka wouldn't start anymore since the Zookeeper is configured invalid. But we can rethrow them here.

In the long run it would also be nice to get this contributed to kafka-junit, so this workaround would be unnecessary. But for now it is easier to keep it that way.